### PR TITLE
Remove experimental x86 float conversion code

### PR DESCRIPTION
**Pull request by 0xdaryl**
Simplify the evaluator by removing experimental paths.

Signed-off-by: Daryl Maier <maier@ca.ibm.com>

### CODE REVIEW

File: compiler/x/codegen/FPTreeEvaluator.cpp [hunk 1]
```diff
@@ -1265,27 +1240,25 @@ TR::Register *OMR::X86::TreeEvaluator::fpConvertToLong(TR::Node *node, TR::Symbo
 
       // For slow conversion only, change the rounding mode on the FPU via its control word register.
       //
-      if (!optLevel)
-         {
-         int16_t fpcw = comp->getJittedMethodSymbol()->usesSinglePrecisionMode() ?
-                           SINGLE_PRECISION_ROUND_TO_ZERO : DOUBLE_PRECISION_ROUND_TO_ZERO;
+      {
```
Comment (fjeremic) [inline compiler/x/codegen/FPTreeEvaluator.cpp]: Is there any need for a new scope here? Can it be removed?

Comment (0xdaryl) [reply compiler/x/codegen/FPTreeEvaluator.cpp]: No reason--just lazy in the refactoring.  I'll fix.  Thanks for keeping me honest!

**Comment by vijaysun-omr** [comment]
Logic seems to have been updated consistently. I will merge once checks pass.

**Comment by fjeremic** [comment]
@genie-omr build xlinux,win

**Pull request merged**